samcli/commands/_utils/options.py

```diff
@@ -82,6 +105,18 @@ def get_or_default_template_file_name(ctx, param, provided_value, include_build)
     :return: Actual value to be used in the CLI
     """
 
+    if provided_value == "-":
+        # "-" is a special value that tells us to read the template from stdin
```

Comment:
This logic isn't specific to deploy. It can be used with any command, but the cleanup step is only done for deploy. Can we clean up the temp files for all commands?